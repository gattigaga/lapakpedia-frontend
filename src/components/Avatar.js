import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Image = styled.img`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  object-fit: cover;
  border-radius: 100%;
`;

const Avatar = props => <Image {...props} />;

Avatar.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
  size: PropTypes.number
};

Avatar.defaultProps = {
  size: 48,
  alt: "Profile"
};

export default Avatar;
