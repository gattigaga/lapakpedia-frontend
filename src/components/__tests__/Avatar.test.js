import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Avatar from "components/Avatar";

describe("Avatar", () => {
  const setup = propOverrides => {
    const props = {
      src: require("assets/images/dummy240x240.png"),
      ...propOverrides
    };

    const wrapper = shallow(<Avatar {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with custom size", () => {
    const { wrapper } = setup({ size: 96 });

    expect(wrapper.props().size).toEqual(96);
  });
});
