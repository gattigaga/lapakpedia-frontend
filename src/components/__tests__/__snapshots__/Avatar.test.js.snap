// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should renders default 1`] = `
<styled.img
  alt="Profile"
  size={48}
  src="dummy240x240.png"
/>
`;
